src/build/buildIcon.ts

```diff
@@ -83,12 +83,16 @@ export function convertIconIfNecessary(options: AppOptions): void {
   }
 
   try {
-    const iconPath = convertToIcns(options.packager.icon);
-    options.packager.icon = iconPath;
-    return;
+    if (!iconIsIcns(options.packager.icon)) {
+      const iconPath = convertToIcns(options.packager.icon);
+      options.packager.icon = iconPath;
+    }
+    if (options.nativefier.tray) {
+      convertToTrayIcon(options.packager.icon);
```

Comment:
My only concern here is for the following scenario:

- Building a mac app
- With a tray icon
- On a non-Mac build machine
- (And may even be passing a `.icns` file to avoid errors)

Will now cause a recoverable error and no icon (which seems like it was the behavior before anyway). Since there is no degradation in functionality, this is fine, but I think we should probably update the `--tray` entry of `api.md` to document this known limitation.

Reply:
Oh didn't think about that. So yeah, thats not possible.